static/js/publisher/pages/RegisterSnap/RegisterSnapForm.tsx

```diff
@@ -0,0 +1,152 @@
+import { ReactNode, useState, Dispatch, SetStateAction } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import {
+  Form,
+  Select,
+  Input,
+  Button,
+  Icon,
+  Row,
+  Col,
+} from "@canonical/react-components";
+
+import type { RegistrationResponse } from "./local-types";
+
+type Props = {
+  isSending: boolean;
+  setIsSending: Dispatch<SetStateAction<boolean>>;
+  setRegistrationResponse: Dispatch<
+    SetStateAction<RegistrationResponse | undefined>
+  >;
+  availableStores: { name: string; id: string }[];
+};
+
+function RegisterSnapForm({
+  isSending,
+  setIsSending,
+  setRegistrationResponse,
+  availableStores,
+}: Props): ReactNode {
+  const [snapName, setSnapName] = useState<string>();
+  const [selectedStore, setSelectedStore] = useState<string>("ubuntu");
+  const [isPrivate, setIsPrivate] = useState<string>("private");
```

Comment:
Name suggests its a boolean. Maybe `privacy` or `privacySetting` or something like that?
Although, for more confusion it's used as a value on input called `public`.

Reply:
Updated.